packages/repository/test/unit/decorator/model-and-relation.decorator.unit.ts

```diff
@@ -96,8 +96,8 @@ describe('model decorator', () => {
     @property()
     customerId: string;
 
-    @belongsTo({target: 'Customer'})
     // TypeScript does not allow me to reference Customer here
```

Comment:
Is this comment still applicable?

Reply:
Yup, I tried out just using the class as the type but the compiler still gave me an error. I would've thought it would be fine considering VSCode didn't tell me that anything was wrong.

Reply:
For posterity: TypeScript emits the following JavaScript code when the property is set to type `Customer`:

```js
__decorate([
    __1.belongsTo(() => Customer),
    __metadata("design:type", Customer)
], Order.prototype, "customer", void 0);

// ...

let Customer = class Customer extends __1.Entity {
};
```

The code setting `design:type` metadata is accessing the value of `Customer` property before it has been set.